test/tests/functional/pbs_execjob_susp_resume.py

```diff
@@ -0,0 +1,546 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2019 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import time
+from tests.functional import *
+from ptl.utils.pbs_logutils import PBSLogUtils
+
+
+class TestPbsExecjobSuspendResume(TestFunctional):
+    """
+    Tests the hook events execjob_postsuspend, execjob_preresume which are
+    called after a job is suspended, and before the job is resumed.
+    """
+    logutils = PBSLogUtils()
+
+    def setUp(self):
+        if len(self.moms) != 2:
+            self.skipTest('test requires three MoMs as input, ' +
+                          'use -p moms=<mom1>:<mom2>')
+        TestFunctional.setUp(self)
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+
+        # execjob_postsuspend hook
+        self.postsuspend_hook_body = """import pbs
+e=pbs.event()
+
+def proc_status(pid):
+    for line in open("/proc/%d/status" % pid).readlines():
+        if line.startswith("State:"):
+            return line.split(":",1)[1].strip().split(' ')[0]
+    return None
```

Comment:
You might want to place the call to open() in a try/except block. If the open fails, PTL will classify it as an error rather than a failure. There are additional instances of proc_status in this file.